communication/mavlink_waypoint_handler.cpp

```diff
@@ -297,7 +297,15 @@ static void waypoint_handler_control_time_out_waypoint_msg(mavlink_waypoint_hand
  *
  * \return  The waypoint in local coordinate frame
  */
-static local_position_t waypoint_handler_set_waypoint_from_frame(waypoint_struct_t* current_waypoint, global_position_t origin);
+static waypoint_local_struct_t waypoint_handler_set_waypoint_from_frame(waypoint_struct_t* current_waypoint, global_position_t origin, dubin_state_t* dubin_state);
```

Comment:
I do not see why you need to change dubin_state when transforming waypoint coordinates from one frame to another.